spring-data-jdbc/src/main/java/org/springframework/data/jdbc/support/RowMapperResultsetExtractorEither.java

```diff
@@ -0,0 +1,76 @@
+package org.springframework.data.jdbc.support;
+
+import org.springframework.jdbc.core.ResultSetExtractor;
+import org.springframework.jdbc.core.RowMapper;
+/**
+ * Represents either a RowMapper or a ResultSetExtractor
+ *
+ * @author Evgeni Dimitrov
+ */
+public class RowMapperResultsetExtractorEither<T> {
```

Comment:
This shouldn't be an either/or decision but rather a component to expose a `ResultSetExtractor`.

Nit: The class name should honor letter casing of `ResultSet` and have an uppercase S in `ResultSet`.